Source/Core/VideoCommon/TextureCacheBase.h

```diff
@@ -199,6 +200,11 @@ class TextureCacheBase
   TexPool::iterator FindMatchingTextureFromPool(const TCacheEntryConfig& config);
   TexAddrCache::iterator GetTexCacheIter(TCacheEntryBase* entry);
 
+  // Return all possible overlapping textures. As addr+size of the textures is not
+  // indexed, this may return false positives.
+  std::pair<TexAddrCache::iterator, TexAddrCache::iterator>
+  FindOverlappingTextures(u32 addr, u32 size_in_bytes);
```

Comment:
Hm? I thought

`std::pair<iter, iter> FindOverlappingTextures(...) const`

will make the this pointer const. So this->textures_by_address will be const, too. So textures_by_address.lower_bound(lower_addr) will return a std::const_iterator.
Your idea sounds good, but we don't want to return const_iterator here.
That is at least what GCC told me :/